fake-server/index.js

```diff
@@ -0,0 +1,39 @@
+const express = require("express")
+const bodyParser = require("body-parser");
+const fakeUsers = require("./fake-users.json");
+const app = express();
+const port = 3000;
+
+
+function cors(req, res, next) {
+    res.header("Access-Control-Allow-Origin", "*");
+    res.header("Access-Control-Allow-Headers", "*");
+    console.log(req.headers);
+    next();
+}
+
+app.use(cors);
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+app.post("/", (req, res) => {
+    const { username, password } = req.body || {};
+    if(!username || !password) {
+        res.status(400).send("Bad request");
+    } else {

```

Comment:
You may also want to consider rewriting these if statements to be a bit more readable rather than nesting the if statements. Maybe something along the lines of:

```
if (!username || !password) {
     res.status(400).send("Bad request");
     return;
}

const user = fakeUsers[username];

if (!user) {
    res.status(403).send("Login failed");
    console.log("Login failed for unknown user", username);
    return;
}

if(password === user.password) {
    res.status(201).send("Login succeeded");
    console.log("Login succeeded for", username);
    return;
}

res.status(403).send("Login failed");
console.log("Login failed for known user", username);
....
```

Reply:
early return? isn't that an antipattern? I'll have a bit of a think about restructuring the logic though.

Reply:
I've not come across it being an antipattern, but I can see why you wouldn't want to use it all scenarios. I quite like the flat structure look but in this case it might not be appropriate as we are not actually returning anything 👍

Reply:
have refactored it out into specific methods with cyclomatic complexity of ~3 each. Hopefully makes it a bit more readable.

Reply:
Yeah that looks nice 👍